index.ts

```diff
@@ -51,35 +85,184 @@ const checkUser = (checkedUser: User, users: Object) => {
+    for (let message of rows) {
+        let userFound = false;
+        let msgUser: any;
+        let takenIds: Array<number> = [];
+        const res = await knex("users").where({id: message.userid});
+        for (let user of res) {
+            takenIds.push(user.id);
+            if (user.id === message.userid) {
+                userFound = true;
+                msgUser = user;
+            }
+        }
+        highestUserId = getHighestFromArr(takenIds);
+        let messageUser: User;
+        if (userFound === true) {
+            messageUser = {
+                username: msgUser.username,
+                userNum: msgUser.usernum,
+                currentChannel: msgUser.currentChannel,
+                id: msgUser.id,
+                messages: []
+            };
+        } else {
+            messageUser = {
+                username: "DeletedUser",
+                userNum: 9999,
+                currentChannel: 0,
+                id: 0,
+                messages: [],
+            };
+        }
+        const newMessage: Message = {
+            user: messageUser,
+            utctime: message.date,
+            message: message.message,
+            channel: message.channel,
+            id: message.id
+        };
+        messages.push(newMessage);
+        messageIds.push(message.id);
+    }
+    highestId = getHighestFromArr(messageIds);
+};
+
+const updateChannelsFromDb = async () => {
+    console.log("Getting channels from database...");
+    try {
+        const rows = await knex.from("channels").select("*");
+        for (const channel of rows) {
+            // console.log(channel);
+            channels[channel.id] = {
+                name: channel.name,
+                id: channel.id,
+                messages: channel.messages
+            };
+        }
+    } catch (err) {
+        console.error(err);
+        throw err;
+    }
+};
+
+const deleteOldUsers = (sNum) => {
+    console.log(sNum + ": Deleting old users...");
+    try {
+        // knex("users")
+        //     .where("snum", "<", sNum)
+        //     .del();
+        knex.raw('delete from users where snum <= ' + sNum + ";")
+            .catch(err => {
+                console.error(err);
+                throw err;
+            });
+        console.log("Users deleted.");
+    } catch (err) {
+        console.error(err);
+        throw err;
+    }
+};
+
+const incrementSNum = () => {
+    return new Promise(async (resolve) => {
+        const rows = await knex("serverid");
+        const sNum = rows[0].snum;
+        try {
+            await knex("serverid").update({snum: Number(Number(sNum) + 1)});
+        } catch (err) {
+            console.error(err);
+            throw err;
+        }
+        resolve(sNum);
+    });
+};
+
+const init = async () => {
+    console.log("Initializing...");
+    const sNum = await incrementSNum();
+    deleteOldUsers(sNum);
+    updateMessagesFromDb();
+    updateChannelsFromDb();
```

Comment:
Either await all of them if you want those to happen in a sequence or await Promise.all if you want them in parallel